ts/src/header-validator/source.test.ts

```diff
@@ -1279,6 +1279,69 @@ const testCases: TestCase[] = [
     ],
   },
 
+  {
+    name: 'trigger-state-cardinality-valid',
+    json: `{"destination": "https://a.test"}`,
+    sourceType: SourceType.event,
+    noteInfoGain: true,
+    vsv: {
+      maxEventLevelChannelCapacityPerSource: {
+        [SourceType.event]: Infinity,
+        [SourceType.navigation]: 0,
+      },
+      maxSettableEventLevelEpsilon: 14,
+      maxCardinalityOfPossibleTriggerStates: 3,
+    },
+    expectedNotes: [
+      {
+        path: [],
+        msg: 'information gain: 1.58',
+      },
+      {
+        path: [],
+        msg: 'number of possible output states: 3',
+      },
+      {
+        path: [],
+        msg: 'randomized trigger rate: 0.0000025',
+      },
+    ],
+  },
+  {
+    name: 'trigger-state-cardinality-invalid',
+    json: `{"destination": "https://a.test"}`,
+    sourceType: SourceType.event,
+    noteInfoGain: true,
+    vsv: {
+      maxEventLevelChannelCapacityPerSource: {
+        [SourceType.event]: Infinity,
+        [SourceType.navigation]: 0,
+      },
+      maxSettableEventLevelEpsilon: 14,
+      maxCardinalityOfPossibleTriggerStates: 2,
+    },
+    expectedErrors: [
+      {
+        path: [],
+        msg: 'number of possible output states: 3 exceeds max trigger-state cardinality (2)',
```

Comment:
Seeing the full message here, I think this would be better expressed as `number of possible output states (3) exceeds max cardinality (2)`, or something similar. The `:` in particular seems off.

Reply:
Should we change them both?

Reply:
The note message is equivalent to `x is y`, whereas this is analogous to `x is 3 exceeds ...`.

Please update.

Reply:
I think it's ok. The part before the colon indicates what this is about. And then the statement seems pretty succinct.

Reply:
Done!